pkg/controller/stacks/install/installjob.go

```diff
@@ -236,11 +243,21 @@ func (jc *stackInstallJobCompleter) createJobOutputObject(ctx context.Context, o
 		if obj.GetNamespace() == "" {
 			obj.SetNamespace(i.GetNamespace())
 		}
+
+		obj.SetOwnerReferences([]metav1.OwnerReference{
+			meta.AsOwner(meta.ReferenceTo(i, i.GroupVersionKind())),
+		})
 	}
 
-	// set an owner reference on the object
-	obj.SetOwnerReferences([]metav1.OwnerReference{
-		meta.AsOwner(meta.ReferenceTo(i, i.GroupVersionKind())),
+	// We want to clean up any installed CRDS when we're deleted. We can't rely
+	// on garbage collection because a namespaced object (StackInstall) can't
+	// own a cluster scoped object (CustomResourceDefinition), so we use labels
+	// instead. TODO(negz): Can obj be anything other than a Stack or a
+	// CustomResourceDefinition here?
```

Comment:
RBACs also created for stacks, where a clusterrole could be owned by a namespaced stack instance which is the same issue. But, AFAIU, it is in stack controller not here.